nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php

```diff
@@ -271,17 +303,15 @@ private function cwpGetPostcodeDistrict(string $postcode) {
 
     // If postcode is a full NI postcode, or just the first part
     // (outward code - e.g. BT1) ...
-    if (preg_match('/^BT[0-9]{1,2}( ?[0-9][A-Z]{2})?$/i', $postcode)) {
-      if (strlen($postcode) > 4) {
-        // Full postcode - remove first 2 and last 3 characters plus any
-        // trailing spaces to get the district number.
-        $postcode_district = rtrim(substr($postcode, 2, -3));
-      }
-      else {
-        // Postcode outward code - just strip of first two 'BT' characters.
-        $postcode_district = substr($postcode, 2);
-      }
 
```

Comment:
Regex validation of the postcode is handled by validatePostcode() instead of here in cwpGetPostcodeDistrict()